datastore/gcloud/aio/datastore/value.py

```diff
@@ -74,6 +78,10 @@ def to_repr(self) -> Dict[str, Any]:
     def _infer_type(self, value: Any) -> TypeName:
         odd_strings = {'newstr', 'unicode'}
         kind = str if value.__class__.__name__ in odd_strings else type(value)
+        # We should instead use Entity as the key in TYPES directly, but that
```

Comment:
As mentioned above, applying this change to `_get_supported_types()` directly would be a bit nicer -- I believe that should eliminate the need for this check.